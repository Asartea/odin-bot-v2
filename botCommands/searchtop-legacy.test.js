const command = require('./searchtop-legacy');

describe('!searchtop', () => {
  describe('regex', () => {
    it.each([
      ['!searchtop query'],
      [' !searchtop query'],
      ['!searchtop query @odin-bot'],
      ['@odin-bot !searchtop query'],
    ])('correct strings trigger the callback', (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['!searchto'],
      ['searchtop'],
      ['!searcht'],
      ['!search top'],
      ['```function("!searchtop", () => {}```'],
      ['!searchingtop'],
      [''],
      [' '],
      [' !'],
      ['@odin-bot ! searchtop'],
      ['!search-top'],
      ['!searc^htop'],
      ['!searchtop!'],
      ['@odin-bot! searchtop'],
      ['https:!!searchtop.com'],
    ])("'%s' does not trigger the callback", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });

    it.each([
      ['Check this out! !searchtop query'],
      ["Don't worry about !searchtop query"],
      ['Hey @odin-bot, !searchtop query'],
      ['!@odin-bot ^ !me !searchtop !tests$*'],
    ])("'%s' - command can be anywhere in the string", (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['@user!searchtop'],
      ["it's about!searchtop"],
      ['!searchtopnow'],
      ['!searchtop!'],
      ['!searchtop*'],
      ['!searchtop...'],
    ])(
      "'%s' - command should be its own word!group - no leading or trailing characters",
      (string) => {
        expect(command.regex.test(string)).toBeFalsy();
      },
    );
  });

  describe('callback', () => {
    it('returns correct output', () => {
      expect(
        command.cb({ content: '!searchtop The Odin Project' }),
      ).toMatchSnapshot();
    });
  });
});
