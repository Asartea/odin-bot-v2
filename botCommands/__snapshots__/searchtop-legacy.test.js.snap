// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`!searchtop callback returns correct output 1`] = `
{
  "embeds": [
    {
      "color": 13407555,
      "description": "Here are the [Google results for TOP lessons containing "The Odin Project"](https://www.google.com/search?q=site:theodinproject.com+The+Odin+Project)",
      "title": "Search TOP",
    },
  ],
}
`;
